include/fastdds/rtps/writer/RTPSWriter.h

```diff
@@ -400,54 +388,47 @@ class RTPSWriter
     LivelinessLostStatus liveliness_lost_status_;
 
     /**
-     * Check if the destinations managed by this sender interface have changed.
-     *
-     * @return true if destinations have changed, false otherwise.
+     * @return Whether the writer is data sharing compatible or not
      */
-    bool destinations_have_changed() const override;
+    bool is_datasharing_compatible() const;
 
-    /**
-     * Get a GUID prefix representing all destinations.
+    /*!
+     * Tells writer the sample can be sent to the network.
+     * This function should be used by a fastdds::rtps::FlowController.
      *
-     * @return When all the destinations share the same prefix (i.e. belong to the same participant)
-     * that prefix is returned. When there are no destinations, or they belong to different
-     * participants, c_GuidPrefix_Unknown is returned.
+     * @param cache_change Pointer to the CacheChange_t that represents the sample which can be sent.
+     * @return Return code.
+     * @note Must be non-thread safe.
      */
-    GuidPrefix_t destination_guid_prefix() const override;
+    virtual DeliveryRetCode deliver_sample_nts(
+            CacheChange_t* cache_change,
+            RTPSMessageGroup& group,
+            LocatorSelectorSender& locator_selector,
+            const std::chrono::time_point<std::chrono::steady_clock>& max_blocking_time) = 0;
 
-    /**
-     * Get the GUID prefix of all the destination participants.
-     *
-     * @return a const reference to a vector with the GUID prefix of all destination participants.
-     */
-    const std::vector<GuidPrefix_t>& remote_participants() const override;
+    virtual LocatorSelectorSender& get_general_locator_selector() = 0;
 
-    /**
-     * Get the GUID of all destinations.
-     *
-     * @return a const reference to a vector with the GUID of all destinations.
-     */
-    const std::vector<GUID_t>& remote_guids() const override;
+    virtual LocatorSelectorSender& get_async_locator_selector() = 0;
```

Comment:
Could we have these three methods on a forward-declared implementation class, injected on the constructor?
We could then get rid of including all the new headers.